src/sorobandata_builder.js

```diff
@@ -102,22 +100,37 @@ export class SorobanDataBuilder {
     return this;
   }
 
+  /**
+   * Appends the given ledger keys to the existing storage access footprint.
+   * @param {xdr.LedgerKey[]} readOnly   read-only keys to add
+   * @param {xdr.LedgerKey[]} readWrite  read-write keys to add
+   * @returns {SorobanDataBuilder} this builder instance
+   */
+  appendFootprint(readOnly, readWrite) {
```

Comment:
is it worth here to check if the `readOnly`/`readWrite` are actually `LedgerKey`s?

Reply:
nah, we pretty much assume sanity in many, **many** places in this codebase. the TypeScript should do a good enough job enforcing this